solver/llbsolver/history.go

```diff
@@ -329,6 +331,47 @@ func (h *HistoryQueue) gc() error {
 	return nil
 }
 
+func (h *HistoryQueue) clearOrphans() error {
+	ctx := context.Background()
+	var records []*controlapi.BuildHistoryRecord
+
+	if err := h.opt.DB.View(func(tx *bolt.Tx) error {
+		b := tx.Bucket([]byte(recordsBucket))
+		if b == nil {
+			return nil
+		}
+		return b.ForEach(func(key, dt []byte) error {
+			var br controlapi.BuildHistoryRecord
+			if err := br.Unmarshal(dt); err != nil {
```

Comment:
can't use `proto` on v0.16 since gogo proto removal:

```
58.40 # github.com/moby/buildkit/solver/llbsolver
58.40 solver/llbsolver/history.go:345:14: undefined: proto
```